factorial.go

```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"fmt"
+)
+
+func calculateFactorial(i uint) (result uint) {
```

Comment:
minor: according to the task description this function should have `factorial` name. No need to fix it here, just for the future.